src/Symfony/Contracts/HttpClient/Test/HttpClientTestCase.php

```diff
@@ -778,4 +778,15 @@ public function testGzipBroken()
         $this->expectException(TransportExceptionInterface::class);
         $response->getContent();
     }
+
+    public function testTransferTimeout()
+    {
+        $client = $this->getHttpClient(__FUNCTION__);
+        $response = $client->request('GET', 'http://localhost:8057/transfer-timeout', [
+            'transfer_timeout' => 0.1,
+        ]);
+
+        $this->expectException(TransportExceptionInterface::class);
```

Comment:
Maybe you can add assertion about the time the request / response took?

Reply:
Totally!